x-pack/plugins/observability_solution/logs_explorer/common/data_views/models/data_view_descriptor.ts

```diff
@@ -58,6 +59,10 @@ export class DataViewDescriptor {
     };
   }
 
+  testAgainstAllowedList(allowedList: string[]) {
```

Comment:
**note**: I opted for an approach where a setting preference is not coupled to this entity to make decisions on the user journey flow. Ideally, we'll conditionally be able to make decisions against a data view based on any setting passed, without affecting the inner shape of this entity.